### PR TITLE
SFR-442 add feedback form

### PR DESCRIPTION
**Pull request by mwbenowitz**
This adds a feedback button for user bug reports/responses to the ResearchNow app. The base for this form was borrowed from Shared Collections Catalog and modified with new styles, a new question and a new integration. It has also been updated to work with our version of React, which is somewhat newer than what is in use on SCC.

This sends the responses to an Airtable spreadsheet, which is easier to work with than a Google Sheet Integration.

This has only a very basic test and I'm not sure if the best thing to do is to add more tests now or have some basic tests in place and build out the test suite in the future.

### CODE REVIEW

File: webpack.config.babel.js [hunk 1]
```diff
@@ -21,6 +24,7 @@ const ENV = process.env.NODE_ENV || 'development';
 
 // Sets appEnv so the the header component will point to the search app on either Dev or Prod
 const appEnv = process.env.APP_ENV ? process.env.APP_ENV : 'production';
+const airtableKey = process.env.AIRTABLE_API_KEY || 'broken :(';
```
Comment (hokei) [inline webpack.config.babel.js]: lol

**Comment by hokei** [comment]
re: tests - I'd mount it and make sure it has all the right fields, and then do a check to make sure when submit is clicked the right methods get called.  
No biggie though, I think it's just as fair to file a ticket to put it in later, rather than have you trudge through enzyme documentation trying to get tests to work.

File: src/client/styles/components/Feedback.scss [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      display: inline-block;
+      width: 0.75rem;
+    }
+
+    .sfr-radio-label {
+      display: inline-block;
+      padding-left: 0.3rem;
+      margin: 0.2rem;
+    }
+  }
```
Comment (hokei) [inline src/client/styles/components/Feedback.scss]: nit

**Comment by hokei** [comment]
I can only tab into the "yes" radio button, not the "no"

File: src/app/components/Feedback/Feedback.jsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    this.openForm = this.openForm.bind(this);
+    this.closeForm = this.closeForm.bind(this);
+    this.deactivateForm = this.deactivateForm.bind(this);
+    this.handleFeedbackChange = this.handleFeedbackChange.bind(this);
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.sendFeedback = this.sendFeedback.bind(this);
+    this.handleRadioChange = this.handleRadioChange.bind(this);
+  }
+
+  onSubmitForm(e) {
```
Comment (hokei) [inline src/app/components/Feedback/Feedback.jsx]: Probably needs some sort of error message if you submit the form without the required field
(I'd argue that we probably don't need to require comments, though.  Like, it's useful even to say "Yes I found what I needed")

File: src/client/styles/components/Feedback.scss [hunk 4]
```diff
@@ -0,0 +1,136 @@
+.feedback {
+    position: static;
+    bottom: 0;
+    right: 0;
+    z-index: 100;
+    text-align: right;
+    background: #54514A;
+    color: #000;
+  
+    @include min-screen(1024px) {
```
Comment (hokei) [inline src/client/styles/components/Feedback.scss]: nit: I agree it's probably not necessary to have the button in mobile, but is 1024 too big of a min-screen?

**Comment by mwbenowitz** [comment]
> I can only tab into the "yes" radio button, not the "no"

Apparently this is the proper behavior for radio button groups: https://stackoverflow.com/questions/14322564/can-you-tab-through-all-radio-buttons. Which is odd to me but the form does behave as this describes

File: src/client/styles/components/Feedback.scss (hunk 4 above)
Comment (mwbenowitz) [reply src/client/styles/components/Feedback.scss]: This pins the form to the bottom of the window and forces it full width. The original code I borrowed this from has this as a variable "tablet-portrait" and 1024 is the best estimate of that value I could come up with.

In testing it out it seems to work well in Chrome at least.

**Comment by hokei** [comment]
Re: radio buttons -- yeah, it is intended.  It also throws me off every time ... sigh

File: src/app/components/Feedback/Feedback.jsx (hunk 3 above)
Comment (mwbenowitz) [reply src/app/components/Feedback/Feedback.jsx]: Added in the new commits I just pushed!

File: src/app/components/Feedback/Feedback.jsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+                  className="feedback-input"
+                  name="sfr-general-feedback"
+                  rows="5"
+                  aria-required="true"
+                  tabIndex="0"
+                  ref={(textarea) => this.feedbackField = textarea}
+                  value={this.state.feedback}
+                  onChange={this.handleFeedbackChange}
+                />
+                <div id="textarea-status-note">{this.state.commentStatus}</div>
```
Comment (hokei) [inline src/app/components/Feedback/Feedback.jsx]: you want `{ this.state.commentStatus && <div id="textarea-status-note">Comment Required</div>}` here
(Conventions - don't want to render this div if you don't have to)

File: src/app/components/Feedback/Feedback.jsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+      </div>
+    );
+  }
+}
+
+Feedback.propTypes = {
+  location: PropTypes.object,
+};
+
+export default Feedback;
```
Comment (hokei) [inline src/app/components/Feedback/Feedback.jsx]: nit

**Pull request merged**